extensions/amp-form/0.1/amp-form.js

```diff
@@ -1206,6 +1211,83 @@ export class AmpForm {
     }
   }
 
+  /**
+   * Initialize form inputs from GET query parameter values
+   * if attribute 'initialize-from-url' is present.
+   * @private
+   */
+  maybeInitializeFromUrl_() {
+    if (!this.form_.hasAttribute('initialize-from-url')) {
+      return;
+    }
+
+    let targetChange;
+    const queryParams = parseQueryString(this.win_.location.search);
+    Object.keys(queryParams).forEach(key => {
+      const field = this.form_./*OK*/ querySelector(`[name="${key}"]`);
+      if (!field) {
+        return;
+      }
+      // Do not interfere with form fields that utilize variable substitutions
+      if (field.hasAttribute('data-amp-replace')) {
+        return;
+      }
+
+      const value = queryParams[key] || '';
+      const type = (field.getAttribute('type') || 'text').toLowerCase();
+      const tag = field.tagName;
+
+      // Fields that can be used to alter layout based on CSS selectors like
+      // :checked are not currently allowed. The form elements below can all
+      // have their values set by DOM API .value without causing style updates.
+      const valueInputTypes = [
```

Comment:
Thank you for the suggestion. I opted to leave these variables scoped to `maybeInitializeFromUrl_` in a7d3dd5 since they are specific to the `initialize-from-url` feature, but will be happy to discuss further if you think other methods could benefit from them.

Reply:
Is this resolved?

Reply:
Yes, thanks.